src/app/students/hooks/useStudents.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState, useEffect, useCallback, createContext, useContext, ChangeEvent, PropsWithChildren } from 'react'
+import { parseCSV } from '@/lib/parseCSV'
+import { Student } from '@/_temp_types/student'
+
+interface SectionOption {
+  label: string;
+  value: string;
+}
+
+interface StudentsContextType {
+  displayStudents: Student[];
+  currentSections: SectionOption[];
+  handleFileChange: (event: ChangeEvent<HTMLInputElement>) => Promise<void>;
```

Comment:
Idk if this is needed, my understanding was that once they load students in, they would be saved to the DB, thus there's no such thing as changing the file, only adding new files (which would perform an upsert in the backend)